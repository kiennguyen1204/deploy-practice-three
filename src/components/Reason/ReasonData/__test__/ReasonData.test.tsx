import { render, screen } from '@testing-library/react';

// components
import ReasonItem, { Props } from 'components/Reason/ReasonData';

const defaultProps: Props = {
  item: {
    icon: <img src='assets/images/default_image.webp' alt='Default icon' />,
    title: 'Default Title',
    description: 'Default description'
  }
};

describe('ReasonItem', () => {
  it('renders the component with default props', () => {
    const { asFragment } = render(<ReasonItem {...defaultProps} />);

    const title = screen.getByText(defaultProps.item.title);
    expect(title).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
