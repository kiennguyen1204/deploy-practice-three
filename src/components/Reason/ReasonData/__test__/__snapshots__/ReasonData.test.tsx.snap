// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReasonItem renders the component with default props 1`] = `
<DocumentFragment>
  <section
    class="item-box"
  >
    <div
      class="icon"
    >
      <img
        alt="Default icon"
        src="/src/assets/images/default_image.webp"
      />
    </div>
    <p
      class="title-box"
    >
      Default Title
    </p>
    <p
      class="item-description"
    >
      Default description
    </p>
  </section>
</DocumentFragment>
`;
